### ISSUE TITLE
Import of Azure AD Application resource fails

### ISSUE THREAD

**Issue by WillooWisp** [pulumi/pulumi-azuread#155]
I try to import the resource below with pulumi import, but it fails with error

error: azuread:index/application:Application resource 'aada-test' has a problem: ExactlyOne: "display_name": one of `display_name,name` must be specified. Examine values at 'Application.DisplayName'.
error: azuread:index/application:Application resource 'aada-test' has a problem: ExactlyOne: "name": one of `display_name,name` must be specified. Examine values at 'Application.Name'.
error: Preview failed: one or more inputs failed to validate
error: preview failed

```
{
    "type": "azuread:index/application:Application",
    "name": "aada-test",
    "id": "<objectId>"
}
```

Doing the same thing with the service principal works.

```
{
    "type": "azuread:index/servicePrincipal:ServicePrincipal",
    "name": "aadsp-test",
    "id": "<objectId>"
}
```

**Comment by WillooWisp** [comment]
Any workarounds in order to be able to import these resources?

**Comment by leezen** [comment]
@WillooWisp Sorry that you ran into this. Have you tried importing programmatically instead of via the CLI? https://www.pulumi.com/docs/guides/adopting/import/#pulumi-import-resource-operation

Doing it this way will ensure that the properties are set as part of the import.

**Comment by WillooWisp** [comment]
Yes using ImportId programatically works for Application, but not a very good solution since everything else is managed outside of the code using import through the CLI.

But a similar problem occurred when importing a ServicePrincipalPasswordResource. The ExactlyOne error is for end_date here instead. 

```
{
    "type": "azuread:index/servicePrincipalPassword:ServicePrincipalPassword",
    "name": "Test SP",
    "id": "11cac421-596d-45e8-85af-18489be2674a/password/221ba95f-3f8b-4e44-bff1-420c30dc1a25"
},

error: azuread:index/servicePrincipalPassword:ServicePrincipalPassword resource 'Test SP' has a problem: ExactlyOne: "end_date": one of `end_date,end_date_relative` must be specified. Examine values at 'ServicePrincipalPassword.EndDate'.
```

Importing it programatically does not work, since it does not find the resource for the ImportId, even though it is the same as above and the CLI import finds the resource but fails on the other problem.

```
AzureAD.ServicePrincipalPassword("Test SP", new AzureAD.ServicePrincipalPasswordArgs
            {
                ServicePrincipalId = "11cac421-596d-45e8-85af-18489be2674a",
                Description = passwordDescription,
                Value = password,
                EndDate="2299-12-30T23:00:00+00:00"
            }, new CustomResourceOptions { ImportId = "11cac421-596d-45e8-85af-18489be2674a/password/221ba95f-3f8b-4e44-bff1-420c30dc1a25" });

error: Preview failed: resource '11cac421-596d-45e8-85af-18489be2674a/password/221ba95f-3f8b-4e44-bff1-420c30dc1a25' does not exist
```

**Comment by leezen** [comment]
That's very odd as we should be going down the same code path for the read whether it's programmatically specified via `ImportId` or via the CLI with `pulumi import`. We'll need to spend a bit of time investigating what's going on here.

**Comment by WillooWisp** [comment]
Any luck in finding a solution to the problem?

**Comment by WillooWisp** [comment]
Hi, anything happening at all to this issue? @leezen
